realm-jni/src/io_realm_internal_table.cpp

```diff
@@ -1219,12 +1219,59 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_Table_nativeGetSortedView(
                 return reinterpret_cast<jlong>(pTableView);
             } CATCH_STD()
         default:
-            ThrowException(env, IllegalArgument, "Sort is currently only supported on Integer, Boolean and Date columns.");
+            ThrowException(env, IllegalArgument, "Sort is currently only supported on integer, boolean, double, float, String, and Date columns.");
             return 0;
     }
     return 0;
 }
 
+JNIEXPORT jlong JNICALL Java_io_realm_internal_Table_nativeGetSortedViewMulti(
+   JNIEnv *env, jobject, jlong nativeTablePtr, jlongArray columnIndices, jbooleanArray ascending)
+{
+    Table* pTable = TBL(nativeTablePtr);
+
+    jsize arr_len = env->GetArrayLength(columnIndices);
+    if (arr_len != env->GetArrayLength(ascending)) {
+        ThrowException(env, IllegalArgument, "Number of column indices and sort orders do not match.");
+        return 0;
+    }
+
+    jlong *long_arr = env->GetLongArrayElements(columnIndices, NULL);
+    jboolean *bool_arr = env->GetBooleanArrayElements(ascending, NULL);
+
+    std::vector<size_t> indices;
```

Comment:
We already know the size so we might as well initialize it with the proper size.